src/app/InteractionModelEngine.h

```diff
@@ -343,5 +343,8 @@ CHIP_ERROR ReadSingleClusterData(const Access::SubjectDescriptor & aSubjectDescr
  */
 CHIP_ERROR WriteSingleClusterData(const Access::SubjectDescriptor & aSubjectDescriptor, ClusterInfo & aClusterInfo,
                                   TLV::TLVReader & aReader, WriteHandler * apWriteHandler);
+
+bool IsClusterDataVersionAllowed(const EndpointId & aEndpointId, const ClusterId & aClusterId, const DataVersion & aDataVersion);
+bool IsClusterDataVersionEqual(const EndpointId & aEndpointId, const ClusterId & aClusterId, const DataVersion & aRequiredVersion);
```

Comment:
Again, please don't pass these by reference.

Followup: we should have a ClusterPath containing endpoint and cluster id that other paths inherit from, and pass it here.

Reply:
https://github.com/project-chip/connectedhomeip/issues/14820